cardano-crypto-class/src/Cardano/Crypto/DSIGN/Bip32Ed25519.hs

```diff
@@ -0,0 +1,202 @@
+
+
+instance Show (SignKeyDSIGN Bip32Ed25519DSIGN) where
+  show (SignKeyBip32Ed25519DSIGN sk) = show $ xPrvToBytes sk
+
+instance ToCBOR (VerKeyDSIGN Bip32Ed25519DSIGN) where
+  toCBOR = encodeVerKeyDSIGN
+  encodedSizeExpr _ = encodedVerKeyDSIGNSizeExpr
+
+instance FromCBOR (VerKeyDSIGN Bip32Ed25519DSIGN) where
+  fromCBOR = decodeVerKeyDSIGN
+
+instance ToCBOR (SignKeyDSIGN Bip32Ed25519DSIGN) where
+  toCBOR = encodeSignKeyDSIGN
+  encodedSizeExpr _ = encodedSignKeyDESIGNSizeExpr
+
+instance FromCBOR (SignKeyDSIGN Bip32Ed25519DSIGN) where
+  fromCBOR = decodeSignKeyDSIGN
+
+instance ToCBOR (SigDSIGN Bip32Ed25519DSIGN) where
+  toCBOR = encodeSigDSIGN
+  encodedSizeExpr _ = encodedSigDSIGNSizeExpr
+
+instance FromCBOR (SigDSIGN Bip32Ed25519DSIGN) where
+  fromCBOR = decodeSigDSIGN
+
+
+-- | Serialise an 'CC.XPrv' to a 'ByteString' (96 bytes).
+--
+-- In @cardano-crypto@, an 'CC.XPrv' was originally serialised using the
+-- following 128-byte binary format:
+--
+-- +---------------------------------+-----------------------+-----------------------+
+-- | Extended Private Key (64 bytes) | Public Key (32 bytes) | Chain Code (32 bytes) |
+-- +---------------------------------+-----------------------+-----------------------+
+--
+-- However, this function serialises an 'CC.XPrv' using a more compact 96-byte
+-- binary format:
+--
+-- +---------------------------------+-----------------------+
+-- | Extended Private Key (64 bytes) | Chain Code (32 bytes) |
+-- +---------------------------------+-----------------------+
+--
+xPrvToBytes :: CC.XPrv -> ByteString
+xPrvToBytes xPrv = privateKeyBytes <> chainCodeBytes
+  where
+    privateKeyBytes :: ByteString
+    privateKeyBytes = BS.take 64 (CC.unXPrv xPrv)
+
+    chainCodeBytes :: ByteString
+    chainCodeBytes = BS.drop 96 (CC.unXPrv xPrv)
+
+-- | Deserialise an 'CC.XPrv' from a 'ByteString' (96 bytes).
+--
+-- In @cardano-crypto@, an 'CC.XPrv' was originally deserialised using the
+-- following 128-byte binary format:
+--
+-- +---------------------------------+-----------------------+-----------------------+
+-- | Extended Private Key (64 bytes) | Public Key (32 bytes) | Chain Code (32 bytes) |
+-- +---------------------------------+-----------------------+-----------------------+
+--
+-- However, this function deserialises an 'CC.XPrv' using a more compact
+-- 96-byte binary format:
+--
+-- +---------------------------------+-----------------------+
+-- | Extended Private Key (64 bytes) | Chain Code (32 bytes) |
+-- +---------------------------------+-----------------------+
+--
+xPrvFromBytes :: ByteString -> Maybe CC.XPrv
+xPrvFromBytes bytes
+    | BS.length bytes /= 96 = Nothing
+    | otherwise = do
+        let (prv, cc) = BS.splitAt 64 bytes
+        pub <- ed25519ScalarMult (BS.take 32 prv)
+        eitherToMaybe $ CC.xprv $ prv <> pub <> cc
+  where
+    eitherToMaybe :: Either a b -> Maybe b
+    eitherToMaybe = either (const Nothing) Just
```

Comment:
Is this not in `Cardano.Prelude`?

Reply:
Nope, it's not. Maybe it should be, though.